### PR TITLE
Fix: Port to GTK+3 and Sugar GTK-toolkit 3

### PR DESCRIPTION
**Pull request by rhl-bthr**
**Fixed in this PR:** Activity ported to GTK3 and Sugar Toolkit 3

**Tested on:** Sugar 0.112. Ubuntu 16.04(rdesktop)

Kindly review

### CODE REVIEW

**Comment by rhl-bthr** [comment]
**Update:**
This PR now also fixes: Upgrade to sugargame v1.2

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #! /usr/bin/env python
 # -*- coding: utf-8 -*-
 
-from sugar.activity import bundlebuilder
+from sugar.3activity import bundlebuilder
```
Comment (quozl) [inline setup.py]: No such module `sugar.3activity`.  Add _test setup.py_ to your checklist for future ports.

File: jump.py [hunk 2]
```diff
@@ -623,7 +627,12 @@ def SuperLooper(self):
...
         rollover_once=0
         run=1
-
+        for event in pygame.event.get():
+            if event.type==pygame.QUIT:
+                return
+            elif event.type == pygame.VIDEORESIZE:
+                pygame.display.set_mode(event.size, pygame.RESIZABLE)
+                break
         pygame.init()
```
Comment (quozl) [inline jump.py]: Call to `pygame.init()` seems redundant, does it work without that once you add `pygame.mixer` to canvas modules?

File: activity.py [hunk 3]
```diff
@@ -26,12 +29,15 @@ def __init__(self, handle):
 
         self.game = jump.SolitaireMain()
         self.build_toolbar()
-        self._pygamecanvas = sugargame.canvas.PygameCanvas(self)
+
+        self.game.canvas = self._pygamecanvas = \
+            sugargame.canvas.PygameCanvas(self,
+                main=self.game.SuperLooper,
+                modules=[pygame.display, pygame.font])
```
Comment (quozl) [inline activity.py]: I've not checked the rest of the code that you haven't changed; but you did change a button that mutes sound, so if the sound uses `pygame.mixer` you might add that module to the list of modules at this line.

File: sugargame/event.py [hunk 4]
```diff
@@ -186,72 +183,76 @@ def _get_mouse_pressed(self):
...
         except pygame.error, e:
-            if str(e) == 'Event queue full':
-                print "Event queue full!"
+            if str(e) == 'video system not initialized':
+                pass
+            elif str(e) == 'Event queue full':
+                logging.error("Event queue full!")
                 pass
             else:
                 raise e
```
Comment (quozl) [inline sugargame/event.py:258]: I've not checked your changes to Sugargame v1.2, but it reminds me that there may be activities with a `sugargame/` that has diverged from the original, adding features or fixing bugs, so if and when you find these please contribute them to the Sugargame repository somehow.

**Review by quozl — COMMENTED** [review]
Thanks.  Not tested.  Reviewed.

**Comment by rhl-bthr** [comment]
Thank you @quozl for pointing my mistakes. I will make sure they don't happen in future ports

**Changes made:**
 - [x] Fixed setup.py import statement
 - [x] Added pygame.mixer to list of modules in `PygameCanvas()` call
 - [x] Removed `pygame.init()` (redundant code)

File: sugargame/event.py (hunk 4 above)
Comment (rhl-bthr) [reply sugargame/event.py:258]: I will contribute those changes to Sugargame as soon as I find them. 👍

File: setup.py (hunk 1 above)
Comment (rhl-bthr) [reply setup.py]: Apologies. Such mistakes won't repeat.

**Pull request merged**